registry/garbagecollect.go

```diff
@@ -96,6 +96,18 @@ func markAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 			return nil
 		})
 
+		if err != nil {
+			// When deleting all tags from a repository the S3 driver also
```

Comment:
While this comment is accurate, an incomplete metadata set can be caused by other scenarios (e.g. a backend failure during upload) and since we should fix or find the root cause of the s3 problem, can you amend the comment slightly?